lib/rspec/expectations/fail_with.rb

```diff
@@ -24,8 +24,7 @@ def fail_with(message, expected=nil, actual=nil)
                                "appropriate failure_message[_when_negated] method to return a string?"
         end
 
-        diff = differ.diff(actual, expected)
-        message = "#{message}\nDiff:#{diff}" unless diff.empty?
+        message = ::RSpec::Matchers::ExpectedsForMultipleDiffs.from(expected).message_with_diff(message, method(:differ), actual)
```

Comment:
I think that `differ` can be passed directly as the 2nd argument -- no need to make it a factory.  It'll be a bit more straight forward and efficient that way.  Honestly, the only reason `differ` isn't memoized is because of cases where the user may re-configure `color` -- in that case, a memoized instance could work in properly.  But here that concern doesn't apply because the user won't have a chance to reconfigure `color` while these objects are in use.